src/System.Management.Automation/engine/Modules/ModuleUtils.cs

```diff
@@ -105,6 +105,8 @@ internal static IEnumerable<string> GetAllAvailableModuleFiles(string topDirecto
                     }
                 }
             }
+
+            firstSubDirs = false;
```

Comment:
`firstSubDirs` indicates the first level of module directories under the module path, right?
If so, it seems this line should be moved to the end of the try block?

Reply:
Or, consider `Directory.GetDirectories` may throw, `firstSubDirs = false` should be set right after the catch blocks.